packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -1390,108 +1392,112 @@ namespace Tpetra {
 
-#ifdef HAVE_TPETRA_DEBUG
-      TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
-        (k_ptrs.extent (0) == 0, std::logic_error,
-         "(StaticProfile \"Optimize Storage\" = "
-         "true branch) After packing, k_ptrs.extent(0) = 0.  This "
-         "probably means that k_rowPtrs_ was never allocated.");
-      if (k_ptrs.extent (0) != 0) {
-        const size_t numOffsets = static_cast<size_t> (k_ptrs.extent (0));
-        const auto valToCheck = ::Tpetra::Details::getEntryOnHost (k_ptrs, numOffsets - 1);
+      if (debug) {
+        const char myPrefix[] = "(StaticProfile \"Optimize Storage\""
+          "=true branch) After packing, ";
         TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
-          (static_cast<size_t> (valToCheck) != k_vals.extent (0),
-           std::logic_error,
-           "(StaticProfile \"Optimize Storage\"=true branch) After packing, "
-           "k_ptrs(" << (numOffsets-1) << ") = " << valToCheck <<
-           " != k_vals.extent(0) = " << k_vals.extent (0) << ".");
-        TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
-          (static_cast<size_t> (valToCheck) != k_inds.extent (0),
-           std::logic_error,
-           "(StaticProfile \"Optimize Storage\"=true branch) After packing, "
-           "k_ptrs(" << (numOffsets-1) << ") = " << valToCheck <<
-           " != k_inds.extent(0) = " << k_inds.extent (0) << ".");
+          (k_ptrs.extent (0) == 0, std::logic_error, myPrefix
+           << "k_ptrs.extent(0) = 0.  This probably means that "
+           "k_rowPtrs_ was never allocated.");
+        if (k_ptrs.extent (0) != 0) {
+          const size_t numOffsets (k_ptrs.extent (0));
+          const auto valToCheck =
+            getEntryOnHost (k_ptrs, numOffsets - 1);
+          TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
+            (size_t (valToCheck) != k_vals.extent (0),
+             std::logic_error, myPrefix <<
+             "k_ptrs(" << (numOffsets-1) << ") = " << valToCheck <<
+             " != k_vals.extent(0) = " << k_vals.extent (0) << ".");
+          TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
+            (size_t (valToCheck) != k_inds.extent (0),
+             std::logic_error, myPrefix <<
+             "k_ptrs(" << (numOffsets-1) << ") = " << valToCheck <<
+             " != k_inds.extent(0) = " << k_inds.extent (0) << ".");
+        }
       }
-#endif // HAVE_TPETRA_DEBUG
     }
     else { // We don't have to pack, so just set the pointers.
       k_ptrs_const = myGraph_->k_rowPtrs_;
       k_inds = myGraph_->k_lclInds1D_;
       k_vals = this->k_values1D_;
 
-#ifdef HAVE_TPETRA_DEBUG
+      if (debug) {
+        const char myPrefix[] =
+          "(StaticProfile \"Optimize Storage\"=false branch) ";
+        TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
+          (k_ptrs_const.extent (0) == 0, std::logic_error, myPrefix
+           << "k_ptrs_const.extent(0) = 0.  This probably means "
+           "that k_rowPtrs_ was never allocated.");
+        if (k_ptrs_const.extent (0) != 0) {
```

Comment:
This condition is always true after throwing if it's false.

Reply:
Thanks @brian-kelley ! :-D  I have some other fixes in the pipe, so I'll add this.